cdap-api/src/main/java/co/cask/cdap/api/dataset/table/RowColumns.java

```diff
@@ -29,50 +31,122 @@
 public abstract class RowColumns<T> {
   private final byte[] row;
 
-  private final List<byte[]> columns;
-
-  public byte[] getRow() {
-    return row;
-  }
-
-  public List<byte[]> getColumns() {
-    return columns;
-  }
+  private List<byte[]> columns;
 
   // key & column as byte[]
 
+  /**
+   * Defines a row, including all columns, unless additional columns are specified.
```

Comment:
The "unless" part is unclear. Maybe better be something like "Defines a row with all columns included. If any of the {@code add} method is called, only columns specified through those {@code add} method calls are included."